File: easy/longest_common_prefix.py
class Solution(object):
  def longestCommonPrefix(self, strs):
    """
    :type strs: List[str]
    :rtype: str
    """
    if len(strs) == 0:
      return ''
    if len(strs) == 1:
      return strs[0]

    temp = list(strs[0])
    for str in strs:
      for i in range(len(temp)):
        if len(str) > i and str[i] == temp[i]:
          continue
        else:
          if i == 0:
            return ''
          else:
            temp = list(str[0:i])
            break
    return ''.join(temp)

File: easy/longest_common_prefix.rb
# @param {String[]} strs
# @return {String}
def longest_common_prefix(strs)
  return '' if strs.length == 0
  return strs.first if strs.length == 1

  temp = strs.first
  strs.each do |str|
    temp.split('').each_with_index do |v, i|
      if str[i] == v
        next
      else
        if i == 0
          return ''
        else
          temp = str[0..(i-1)]
          break
        end
      end
    end
  end
  temp
end
